core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -293,22 +301,23 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
 
     while (message ne null) {
       message match {
+        case FiberMessage.Stateful(onFiber) =>
```

Comment:
Most common message we'll get during synchronous evaluation is this, so we check for it first